sceptre/environment.py

```diff
@@ -74,20 +72,25 @@ def __repr__(self):
         )
 
     @staticmethod
-    def _check_env_path_valid(path):
+    def _validate_path(path):
         """
+        Converts backslashes to forward slashes.
         Raises an InvalidEnvironmentPathError if the path has a leading or
         trailing slash.
 
         :param path: A directory path
         :type path: str
         :raises: sceptre.exceptions.InvalidEnvironmentPathError
+        :returns: A normalised path with forward slashes.
+        :returns: string
         """
+        path = path.replace('\\', '/')
```

Comment:
Do we need to worry about environment paths that contain single letter directories which happen to form special chars?

e.g.
```
In [12]: "\a\n\c".replace("\\", "/")
Out[12]: '\x07\n/c'
```